legate/core/legion.py

```diff
@@ -241,12 +243,18 @@ def set_point(self, p):
             self.point.dim = 1
             self.point.point_data[0] = p
 
-    def raw(self):
+    def raw(self) -> Any:
         return self.point
 
 
-class Rect(object):
-    def __init__(self, hi=None, lo=None, exclusive=True, dim=None):
+class Rect:
+    def __init__(
+        self,
+        hi: Optional[list[float]] = None,
```

Comment:
Does `list` mean literally a `list` object or does it encompass other iterables? I remember somewhere I passed a `Shape` object to construct `Rect`.

Reply:
`list` means `list`. It is the the newer replacement for `typing.List` If other iterable types are expected then `Iterable[float]` is probably better

Reply:
Let's make it `Sequence` then. I don't see a reason to limit the type to be just `list` here.

Reply:
https://github.com/nv-legate/legate.core/pull/158/commits/0daeb173635d99e271aa7254549737d776af414b